src/main/java/org/openjdk/jextract/impl/HeaderFileBuilder.java

```diff
@@ -272,16 +291,16 @@ private void emitGlobalSetter(String segmentConstant, String vhConstant, String
     public String emitGlobalSegment(String layout, String javaName, String nativeName, Declaration declaration) {
         String mangledName = mangleName(javaName, MemorySegment.class);
         incrAlign();
-        appendLines(STR."""
-            private static final MemorySegment \{mangledName} = RuntimeHelper.lookupGlobalVariable(\"\{nativeName}\", \{layout});
-
-            """);
         if (declaration != null) {
             emitDocComment(declaration);
         }
         appendLines(STR."""
             \{MEMBER_MODS} MemorySegment \{mangledName}() {
-                return RuntimeHelper.requireNonNull(\{mangledName}, \"\{javaName}\");
+                class Holder {
```

Comment:
I was ambivalent on this - but it's probably better to make it consistent with the strategy we use for functions, as you have done.